src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -462,6 +463,31 @@ public static InteractItemEvent callInteractItemEventSecondary(EntityPlayer play
         return event;
     }
 
+    public static boolean callDestructEntityEvent(Entity entity, Cause cause) {
+        if (((net.minecraft.entity.Entity) entity).isDead && (!(entity instanceof EntityLivingBase) || entity instanceof EntityArmorStand)) {
+            MessageChannel originalChannel = MessageChannel.TO_NONE;
+            Optional<User> sourceCreator = Optional.empty();
```

Comment:
That's meant to be the creator of the entity - it's how it used to be done. The CauseTracker part of it is needed for it to be complete.